be/src/util/mysql_row_buffer.cpp

```diff
@@ -107,7 +107,11 @@ MysqlRowBuffer<is_binary_format>::~MysqlRowBuffer() {
 template <bool is_binary_format>
 void MysqlRowBuffer<is_binary_format>::open_dynamic_mode() {
     if (!_dynamic_mode) {
-        *_pos++ = NEXT_EIGHT_BYTE;
+        // if _pos now exactly at the end of _buf memory,
+        // we should reserve 1 byte for _dynamic_mode flag byte to avoid *pos = 254
+        // cause _dynamic_mode flag byte be overwritten
+        reserve(1);
+        *_pos++ = NEXT_EIGHT_BYTE; // *_pos = 254 ; _pos++
```

Comment:
Can you check that is there any more place should reserve() be called before *_pos++ ?

Reply:
I checked it, in be/src/util/mysql_row_buffer.cpp we have only one place to make *_pos++ 
and I also checked ```push_xxx``` func with reserve func or not , and anything else have I missed ？